community/kernel/src/main/java/org/neo4j/kernel/impl/api/StateHandlingStatementOperations.java

```diff
@@ -1651,4 +1738,53 @@ public String relationshipLegacyIndexRemoveConfiguration( KernelStatement statem
     }
     // </Legacy index>
 
+    private PrimitiveIntIterator getPropertyKeys( EntityCursor entityCursor )
+    {
+        PrimitiveIntStack keys = new PrimitiveIntStack( KEY_LIST_INITIAL_SIZE );
+        try ( PropertyCursor properties = entityCursor.properties() )
+        {
+            while ( properties.next() )
+            {
+                keys.push( properties.propertyKeyId() );
+            }
+        }
+
+        return keys.iterator();
+    }
+
+    private boolean hasProperty( EntityCursor entityCursor, EntityType type, long nodeId, int propertyKeyId )
+            throws EntityNotFoundException
+    {
+        if ( !entityCursor.next() )
+        {
+            throw new EntityNotFoundException( type, nodeId );
+        }
+
+        try ( PropertyCursor cursor = entityCursor.properties() )
+        {
+            return cursor.seek( propertyKeyId );
+        }
+    }
+
+    private Object getProperty( EntityCursor entityCursor,
+            EntityType type,
+            long id,
+            int propertyKeyId ) throws EntityNotFoundException
+
+    {
+        if ( !entityCursor.next() )
+        {
+            throw new EntityNotFoundException( type, id );
+        }
```

Comment:
Entity existence checks like this are very common in this class. Maybe extract method?